--- COLAMD/Source/colamd.c~	2006-12-13 13:46:38.000000000 +0200
+++ COLAMD/Source/colamd.c	2007-03-27 01:17:49.118259378 +0300
@@ -685,7 +685,7 @@
 /* ========================================================================== */
 
 /* define UF_long */
-#include "UFconfig.h"
+#include <UFconfig.h>
 
 #ifdef DLONG
 
--- COLAMD/Source/colamd.h~	2006-12-13 13:46:09.000000000 +0200
+++ COLAMD/Source/colamd.h	2007-03-27 01:18:06.598650636 +0300
@@ -139,7 +139,7 @@
 /* ========================================================================== */
 
 /* define UF_long */
-#include "UFconfig.h"
+#include <UFconfig.h>
 
 size_t colamd_recommended	/* returns recommended value of Alen, */
 				/* or 0 if input arguments are erroneous */
--- COLAMD/Source/colamd_l_example.c~	2006-12-13 13:47:22.000000000 +0200
+++ COLAMD/Source/colamd_l_example.c	2007-03-27 01:18:17.698899095 +0300
@@ -44,7 +44,7 @@
 #define B_N 5
 
 /* define UF_long */
-#include "UFconfig.h"
+#include <UFconfig.h>
 
 int main (void)
 {
--- COLAMD/Makefile~	2007-03-27 01:18:43.219470348 +0300
+++ COLAMD/Makefile	2007-03-27 01:18:44.569500568 +0300
@@ -1,9 +1,7 @@
 
 default: libcolamd.a colamd_example colamd_l_example
 
-include ../UFconfig/UFconfig.mk
-
-I = -I../UFconfig
+include /usr/share/misc/UFconfig.mk
 
 colamd_example: colamd_example.c libcolamd.a
 	$(CC) $(CFLAGS) $(I) -o colamd_example colamd_example.c libcolamd.a -lm
