diff -ur COLAMD/Demo/Makefile COLAMD-shared/Demo/Makefile
--- COLAMD/Demo/Makefile	2007-12-02 15:48:48.000000000 +0100
+++ COLAMD-shared/Demo/Makefile	2007-12-02 15:48:31.000000000 +0100
@@ -20,12 +20,12 @@
 dist:
 
 colamd_example: colamd_example.c library
-	$(C) -o colamd_example colamd_example.c ../Lib/libcolamd.a -lm
+	libtool --tag=CC --mode=link $(CC) $(CFLAGS) $(I) -o colamd_example colamd_example.c ../Lib/libcolamd.la -lm
 	- ./colamd_example > my_colamd_example.out
 	- diff colamd_example.out my_colamd_example.out
 
 colamd_l_example: colamd_l_example.c library
-	$(C) -o colamd_l_example colamd_l_example.c ../Lib/libcolamd.a -lm
+	libtool --tag=CC --mode=link $(CC) $(CFLAGS) $(I) -o colamd_l_example colamd_l_example.c ../Lib/libcolamd.la -lm
 	- ./colamd_l_example > my_colamd_l_example.out
 	- diff colamd_example.out my_colamd_example.out
 
diff -ur COLAMD/Lib/Makefile COLAMD-shared/Lib/Makefile
--- COLAMD/Lib/Makefile	2007-12-02 15:48:48.000000000 +0100
+++ COLAMD-shared/Lib/Makefile	2007-12-02 15:47:12.000000000 +0100
@@ -2,7 +2,8 @@
 # COLAMD Makefile
 #-------------------------------------------------------------------------------
 
-default: libcolamd.a
+default: libcolamd.la
+libdir	:= /usr/lib
 
 include /usr/share/misc/UFconfig.mk
 
@@ -13,15 +14,20 @@
 SRC = ../Source/colamd.c ../Source/colamd_global.c
 
 # creates libcolamd.a, a C-callable COLAMD library
-libcolamd.a:  $(SRC) $(INC)
-	$(CC) $(CFLAGS) $(I) -c ../Source/colamd_global.c
-	$(CC) $(CFLAGS) $(I) -c ../Source/colamd.c
-	$(CC) $(CFLAGS) $(I) -c ../Source/colamd.c -DDLONG -o colamd_l.o
-	$(AR) libcolamd.a colamd.o colamd_l.o colamd_global.o
+libcolamd.a: libcolamd.la
+libcolamd.la:  $(SRC) $(INC)
+	libtool --tag=CC --mode=compile $(CC) $(CFLAGS) $(I) -c ../Source/colamd_global.c
+	libtool --tag=CC --mode=compile $(CC) $(CFLAGS) $(I) -c ../Source/colamd.c
+	libtool --tag=CC --mode=compile $(CC) $(CFLAGS) $(I) -c ../Source/colamd.c -DDLONG -o colamd_l.o
+	libtool --tag=CC --mode=link $(CC) -o libcolamd.la -rpath $(libdir) colamd.o colamd_l.o colamd_global.o -lm
+
+install: libcolamd.la
+	install -d $(DESTDIR)$(libdir)
+	libtool --mode=install cp libcolamd.la $(DESTDIR)$(libdir)/libcolamd.la
 
-ccode: libcolamd.a
+ccode: libcolamd.la
 
-library: libcolamd.a
+library: libcolamd.la
 
 clean:
 	- $(RM) $(CLEAN)
@@ -29,4 +35,4 @@
 purge: distclean
 
 distclean: clean
-	- $(RM) libcolamd.a
+	- $(RM) libcolamd.la
